### PR TITLE
Update the FEEL documentation for version 8.3

### LINKED ISSUE #2650: Update the FEEL documentation for version 8.3
## Description 

Camunda 8.3 includes a new version of the FEEL-Scala engine [1.17.0](https://github.com/camunda/feel-scala/releases/tag/1.17.0). 

```[tasklist]
### Tasks
- [x] Update the FEEL docs
- [x] Mention the new error handling and the impact on incidents in the update guide
- [x] Mention the change of the `is defined()` function in the update guide
- [x] Update incident docs
```

### PR DESCRIPTION
**Pull request by saig0**
## Description

Camunda 8.3 uses a new version of FEEL-Scala `1.17.0`. 

- Copy the FEEL documentation from FEEL-Scala to add the updated content.
- Mention the changes in the update guide to version `8.3`.
- Update the incident-related pages.
- Update the expressions page to remove duplicated content.

closes #2650 

## When should this change go live?

- [x] This change is not yet live and should not be merged for the release of version 8.3 
- [ ] There is no urgency with this change.
- [x] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by saig0** [comment]
@christinaausley I added content to the update guide for version `8.3`. I mention changes what are relevant for expressions in BPMN processes. 

Since the update guide moved to `Self-Managed > Operational guides > Update guide`, I'm not sure if our users will find the content. These changes are not only relevant for self-managed but also for SaaS.

File: docs/components/modeler/feel/language-guide/feel-functions.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ contains(string: "me@camunda.com", match: ".de")
 
 :::info GOOD TO KNOW
 
-If an expression invokes a built-in function but the argument types don't match the function
-signature, the invocation returns `null`.
+The invocation returns `null` if the no function exists with the given name, or if the argument
```
Comment (nicpuppa) [inline docs/components/modeler/feel/language-guide/feel-functions.md:25]:
```suggestion
The invocation returns `null` if no function exists with the given name, or if the argument
```

**Review by nicpuppa — APPROVED** [review]
LGTM 🚀 

Just a small suggestion 👀

**Comment by akeller** [comment]
> Since the update guide moved to `Self-Managed > Operational guides > Update guide`, I'm not sure if our users will find the content. These changes are not only relevant for self-managed but also for SaaS.

I agree here on the visibility concern. What about adding this to the **Announcements** page and linking to the FEEL docs rather than using the **Update guide**?

**Comment by christinaausley** [comment]
> > Since the update guide moved to `Self-Managed > Operational guides > Update guide`, I'm not sure if our users will find the content. These changes are not only relevant for self-managed but also for SaaS.
> 
> I agree here on the visibility concern. What about adding this to the **Announcements** page and linking to the FEEL docs rather than using the **Update guide**?

Agreed here as well. @saig0 Let me know your thoughts and if I can assist, or if you are just looking for general review here 👍

**Comment by saig0** [comment]
> What about adding this to the Announcements page and linking to the FEEL docs rather than using the Update guide?

@akeller thank you for the hint. I looked at the Announcements page. But I think that the content doesn't match the existing announcements. 

The content is relevant for users updating from version 8.2 to 8.3. It contains a warning and how to mitigate possible issues. 

In the past, this kind of information was part of the update guide (for C7 and C8). So, users might expect the content to be in the update guide. And, the update guide was linked from previous release blog posts when updating from an older version.

I would prefer to keep this information in the update guide. 

IMO, the update guide under the Self-Managed section is hard to find.

**Comment by saig0** [comment]
@christinaausley I'm requesting your general review. :smiley: :+1:

**Review by nicpuppa — DISMISSED** [review]

**Comment by akeller** [comment]
@saig0, since most of the content in the **Update guide** was for self-managed users, we moved it. We can consider an **Update guide** for SaaS users since I agree with the visibility aspect. I'll add this to our backlog.

File: docs/components/concepts/expressions.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ Additionally, the following attributes of BPMN elements can define an expression
 - User task: [assignee](/components/modeler/bpmn/user-tasks/user-tasks.md#assignments), [candidateGroups](/components/modeler/bpmn/user-tasks/user-tasks.md#assignments)
 - Call activity: [process id](/components/modeler/bpmn/call-activities/call-activities.md#defining-the-called-process)
 
+And many more.
```
Comment (christinaausley) [inline docs/components/concepts/expressions.md:26]: This is a bit vague. Can we link to these, or be more specific?

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-boolean.md [hunk 3]
```diff
@@ -47,11 +46,93 @@ is defined(1)
...
+// false
 
 is defined(x)
 // false - if no variable "x" exists
 
 is defined(x.y)
 // false - if no variable "x" exists or it doesn't have a property "y"
 ```
+
+:::caution Breaking change
```
Comment (christinaausley) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-boolean.md:58]: As this is a breaking change, should this be noted anywhere else? Like announcements, or update material?

Comment (saig0) [reply docs/components/modeler/feel/builtin-functions/feel-built-in-functions-boolean.md:58]: I mentioned this change in the update guide [here](https://github.com/camunda/camunda-docs/pull/2658/files#diff-0da863f7f864baad0f87fef1747fb29bcda9f68da564fcec08bd4e2544f1782cR82-R92). :+1: 

For me, the update guide seems to be the best place for it. It is mainly relevant to updating from a previous version.

For an announcement, the change is probably too small and too much detail.

File: docs/components/concepts/expressions.md (hunk 2 above)
Comment (saig0) [reply docs/components/concepts/expressions.md:26]: True. :sweat_smile: 

Currently, the text mentions a few attributes but not all of them. I don't see value in having a complete list of attributes here. And, it is hard to maintain when adding new features. 

What do you about removing the whole list? 

Instead, we could mention one attribute as an example.

Comment (christinaausley) [reply docs/components/concepts/expressions.md:26]: I agree with this, can always link to additional documentation as well.

**Comment by christinaausley** [comment]
@saig0 Thanks for addressing my comments! Once resolved, I can approve and merge 👍

**Comment by saig0** [comment]
@christinaausley I addressed your comment as discussed. Please have a look. :cookie:

**Review by christinaausley — APPROVED** [review]
Nice change! Feel free to merge when ready.

**Pull request merged**